Asterisk.2013/Asterisk.NET/Manager/ManagerConnection.cs

```diff
@@ -1106,11 +1121,29 @@ private void internalEventHandler(object sender, ManagerEvent e)
 					#endregion
 
 					case 62:
-						if (ConnectionState != null)
-						{
-							ConnectionState(this, (ConnectionStateEvent)e);
-							return;
-						}
+						if (Connect != null)

```

Comment:
Ok, I didn't know that...
`if(Connect != null)`
...checks if event has a registered handler.
You can discard this code. :)